File: Proyecto LP/mi-servidor/Backend/src/main/java/com/miapp/mi_servidor/Enums/Transmision.java
package com.miapp.mi_servidor.Enums;

/**
 *
 * @author Kathy
 */
public enum Transmision {
    MANUAL("Manual"),
    AUTOMATICA("Automatica"),
    CVT("Continuamente Variable"),
    DSG("Doble Embrague"),
    AMT("Manual Automatizada"),
    MANUAL_SECUENCIAL("Manual Secuencial"),
    EVT("Electronica Variable"),
    HIDRAULICA("Hidraulica");

    private final String displayName;

    Transmision(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    // Método fromValue
    public static Transmision fromValue(String value) {
        for (Transmision transmision : Transmision.values()) {
            if (transmision.getDisplayName().equalsIgnoreCase(value)) {
                return transmision;
            }
        }
        throw new IllegalArgumentException("Valor inválido para el enum Transmision: " + value);
    }
}

File: Proyecto LP/mi-servidor/Frontend/lib/Enums/Transmision.dart
enum Transmision {
  manual("Manual", "MANUAL"),
  automatica("Automatica", "AUTOMATICA"),
  cvt("Continuamente Variable", "CVT"),
  dsg("Doble Embrague", "DSG"),
  amt("Manual Automatizada", "AMT"),
  manualSecuencial("Manual Secuencial", "MANUAL_SECUENCIAL"),
  evt("Electronica Variable", "EVT"),
  hidraulica("Hidraulica", "HIDRAULICA");

  final String displayName;
  final String backendValue;

  const Transmision(this.displayName, this.backendValue);

  static String? getBackendValue(String? displayName) {
    return Transmision.values.firstWhere(
      (transmision) => transmision.displayName == displayName,
      orElse: () => throw Exception("Valor inválido para el enum Transmision: $displayName"),
    ).backendValue;
  }
}